tracer/src/Datadog.Trace/ProcessTags.cs

```diff
@@ -0,0 +1,68 @@
+// <copyright file="ProcessTags.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Datadog.Trace.Processors;
+
+namespace Datadog.Trace;
+
+internal static class ProcessTags
+{
+    public const string EntrypointName = "entrypoint.name";
+    public const string EntrypointBasedir = "entrypoint.basedir";
+    public const string EntrypointWorkdir = "entrypoint.workdir";
+
+    private static Lazy<string> _lazySerializedTags = new(GetSerializedTags);
+
+    public static string SerializedTags
+    {
+        get => _lazySerializedTags.Value;
+    }
+
+    private static Dictionary<string, string> LoadBaseTags()
+    {
+        var tags = new Dictionary<string, string>();
+
+        var entrypointFullName = Assembly.GetEntryAssembly()?.EntryPoint?.DeclaringType?.FullName;
```

Comment:
if I'm not mistaken there are at least 2 different `entrypoint.type` in dotnet, a binary or a .dll
similarly in java the split is done with entrypoint.type:jar_name/main_class

Reply:
yes, I was thinking of adding those dotnet specific details in an other PR, where we could discuss the specifics.
I can see several entry point types : 
 - Console app
 - GUI (winforms/WPF)
 - ASP net webapp
 - windows service
 - Azure function
 - triggered by another library ? (like tests ?)
 - Xamarin mobile app ?

Not sure the distinction exe/dll is super relevant, because a dll can behave the same way as an exe when launched with `dotnet run`.